protocol/common/message_sender.go

```diff
@@ -320,17 +324,71 @@ func (s *MessageSender) sendCommunity(
 	// Check if it's a key exchange message. In this case we send it
 	// to all the recipients
 	if rawMessage.CommunityKeyExMsgType != KeyExMsgNone {
-		keyExMessageSpecs, err := s.protocol.GetKeyExMessageSpecs(rawMessage.HashRatchetGroupID, s.identity, rawMessage.Recipients, rawMessage.CommunityKeyExMsgType == KeyExMsgRekey)
-		if err != nil {
-			return nil, err
+		forceRekey := rawMessage.CommunityKeyExMsgType == KeyExMsgRekey
+		// If rekeycompatibility is on, we always
+		// want to execute below, otherwise we execute
+		// only when we want to fill up old keys to a given user
+		if RekeyCompatibility || !forceRekey {
+			keyExMessageSpecs, err := s.protocol.GetKeyExMessageSpecs(rawMessage.HashRatchetGroupID, s.identity, rawMessage.Recipients, forceRekey)
+			if err != nil {
+				return nil, err
+			}
+
+			for i, spec := range keyExMessageSpecs {
+				recipient := rawMessage.Recipients[i]
+				_, _, err = s.sendMessageSpec(ctx, recipient, spec, messageIDs)
+				if err != nil {
+					return nil, err
+				}
+			}
 		}
 
-		for i, spec := range keyExMessageSpecs {
-			recipient := rawMessage.Recipients[i]
-			_, _, err = s.sendMessageSpec(ctx, recipient, spec, messageIDs)
+		if forceRekey {
+
+			var ratchet *encryption.HashRatchetKeyCompatibility
+			// We have just rekeyed, pull the latest
+			if forceRekey && RekeyCompatibility {
```

Comment:
```suggestion
			if RekeyCompatibility {
```
`forceRekey` is redundant